magento/magento2
# Improve Extensibility of Service URL Generation

### Description (*)
There are some legitimate cases for wanting to customise this function. Let's encourage people to do so in a more maintainable way i.e. via plugins instead of preferences.

### Fixed Issues
#37404 

### Manual testing scenarios (*)
1. Create [a plugin](https://developer.adobe.com/commerce/php/development/components/plugins/) for the `Magento\Store\Model\Argument\Interpreter\ServiceUrl::getServiceUrl()` function
2. Observe that the plugin code executes as expected

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37404: Improve Extensibility of Service URL Generation
